ext/mixed/js/core.js

```diff
@@ -288,6 +288,42 @@ const yomichan = (() => {
             this.trigger('orphaned', {error});
         }
 
+        getTemporaryListenerResult(eventHandler, userCallback, timeout=30000) {
+            let resolved = false;
+            let resolve;
+            let reject;
+            const listenerPromise = new Promise((_resolve, _reject) => {
+                resolve = _resolve;
+                reject = _reject;
+            });
+
+            if (eventHandler === chrome.runtime.onMessage) {
+                const runtimeMessageCallback = ({action, params}, sender, sendResponse) => {
+                    const cleanupResolve = (value) => {
+                        resolved = true;
+                        eventHandler.removeListener(runtimeMessageCallback);
+                        sendResponse();
+                        resolve(value);
+                    };
+
+                    setTimeout(() => {
```

Comment:
Store the timer ID and clear the timer in cleanupResolve also (if it hasn't already completed). Clears the timer resource earlier, probably better for memory.

Reply:
https://github.com/FooSoft/yomichan/pull/417/commits/894c13cc32d70de4a3ca5b78a7ea5fb04b5f791f

Reply:
Btw, I wonder if `clearTimeout` actually needs to check if the timer has been cleared. From what I can gather it's idempotent and you can even call it on ID's that never existed without warnings. Performance wise an `if (bool)` should be faster than a function call (when it's `false`), but it adds a few lines and some indentation.